model.go

```diff
@@ -13,22 +13,29 @@ type TableReader interface {
 }
 
 // ModelDTO - the data transfer object for a model
-type ModelDTO interface {
+type ModelDTO struct {
+	IsDTO
+}
+
+// IsDTO marks an object as DTO
+type IsDTO interface {
 	getID() string
 }
 
-// NewModel - Factory for generating models
+// NewModel - factory for generating models
 func NewModel(t TableReader) Model {
 	return Model{t}
 }
 
+// MakeDTO - factory to generate a ModelDTO
+func MakeDTO(t IsDTO) ModelDTO {
+	return ModelDTO{t}
+}
+
 // CreateRecord is used to insert a new row into a table
 func (m Model) CreateRecord(d ModelDTO) (ModelDTO, error) {
 	rows := m.getTableData().([]ModelDTO)
-	rows = append(rows, d)
+	rows = append((rows), d)
```

Comment:
It appears the rows returned here is of type `[]ModelDTO`, and so updates to the `rows` variable is not reflected on the [users](https://github.com/jsamchineme/go-api/blob/fix-typing-in-CreateRecord/user.go#L18) variable.